city/crud.py

```diff
@@ -0,0 +1,51 @@
+from sqlalchemy.orm import Session
+
+from city import models, schemas
+
+
+def create_city(db: Session, city: schemas.CityCreateUpdate) -> models.City:
+    db_city = models.City(**city.model_dump())
+    db.add(db_city)
+    db.commit()
+    db.refresh(db_city)
+
+    return db_city
+
+
+def get_city(db: Session, city_id: int) -> models.City:
+    return db.query(models.City).filter(models.City.id == city_id).first()
+
+
+def get_city_list(
+    db: Session, skip: int = 0, limit: int = 10
+) -> list[models.City]:
+    return db.query(models.City).offset(skip).limit(limit).all()
+
+
+def get_all_cities(db: Session) -> list[models.City]:
+    return db.query(models.City).all()
+
+
+def update_city(
+    db: Session, city_id: int, city: schemas.CityCreateUpdate
+) -> models.City:
+    db_city = db.query(models.City).filter(models.City.id == city_id).first()
```

Comment:
In the `update_city` function, ensure that the `city_id` exists before attempting to update. Although the current code checks if `db_city` is not None, it might be useful to raise an exception or return a specific response if the city is not found.